src/Data/List/Compat.hs

```diff
@@ -70,4 +129,15 @@ uncons                  :: [a] -> Maybe (a, [a])
 uncons []               = Nothing
 uncons (x:xs)           = Just (x, xs)
 
+-- | A strictly accumulating version of 'scanl'
+{-# NOINLINE [1] scanl' #-}
+scanl'           :: (b -> a -> b) -> b -> [a] -> [b]
```

Comment:
For some reason it also lacks the `@since` annotation, maybe it's not in the changelog too (haven't checked).

I added it to the readme for now, but not sure how much effort we should put into maintaining it.  Once `base-compat` is pretty complete, it may not be that crucial anymore.  Or maybe we want to switch to a list of things that are not covered, or has limitations later on.